test/verify_extra_test.c

```diff
@@ -251,9 +252,14 @@ static int do_test_purpose(int purpose, int expected)
     if (!TEST_true(X509_STORE_CTX_init(ctx, NULL, eecert, untrusted)))
         goto err;
 
-    if (!TEST_true(X509_STORE_CTX_set_purpose(ctx, purpose)))
+    if (!TEST_true(X509_STORE_CTX_set_purpose_ex(ctx, purpose, override)))
         goto err;
 
+    if (purpose2 != 0) {
+        if (!TEST_true(X509_STORE_CTX_set_purpose_ex(ctx, purpose2, override)))
+            goto err;
+    }
```

Comment:
These tests are too weak.
They just check that the call is successful, but not that the right value is set (let alone that the other values are unaffected).